src/context.rs

```diff
@@ -0,0 +1,122 @@
+use core::marker::PhantomData;
+use {ffi, types::{c_uint, c_void}, Error, Secp256k1, };
+
+#[cfg(feature = "std")]
+pub use self::std_only::*;
+
+/// A trait for all kinds of Context's that let's you define the exact flags and a function to deallocate memory.
+/// * DO NOT * implement it for your own types.
+pub unsafe trait Context {
```

Comment:
I think we should just drop `pub` rather than posting a warning :)

Reply:
I tried that, sadly if any of the public functions rely of this you can't make it private